src/projects/TodoList/index.js

```diff
@@ -43,20 +46,45 @@ class TodoList extends React.Component {
     // taskListToBeAltered.splice(i, 1, alteredTask);
     this.setState({ todos: taskListToBeAltered });
   }
+  showEditField(i) {
+    let taskListToBeAltered = this.state.todos;
+    let alteredTask = taskListToBeAltered[i];
+    alteredTask.isEditing = !alteredTask.isEditing;
+    this.setState({ todos: taskListToBeAltered });
+    this.setState({ editedItem: alteredTask.task });
```

Comment:
You can merge these setState calls into one - remember, the object you pass into the setState function is merged with the state object, so you can have 2 key:value pairs at the same time

```js
this.setState({
  todos: taskListToBeAltered,
  editedItem: alteredTask.task
})
```

Reply:
ah shit yeah you're right, ive done that before and forgot